git-changed-files.js

```diff
@@ -62,12 +62,36 @@ const isFileIgnored = (workingDirectory, file) => {
 /**
  * This lists the files that have changed when compared to `base` (a git ref),
  * limited to the files that are a descendent of `cwd`.
+ * It also respects '.gitattributes', filtering out files that have been marked
+ * as "binary" or "linguist-generated=true".
  */
 const gitChangedFiles = async (
     base /*:string*/,
     cwd /*:string*/,
 ) /*: Promise<Array<string>>*/ => {
     cwd = path.resolve(cwd);
+
+    // Github actions jobs can run the following steps to get a fully accurate
+    // changed files list. Otherwise, we fallback to a simple diff between the
+    // current and base branch, which might give false positives if the base
+    // is ahead of the current branch.
+    //
+    //   - name: Get All Changed Files
+    //     uses: jaredly/get-changed-files@absolute
+    //     id: changed
+    //     with:
+    //       format: 'json'
+    //       absolute: true
```

Comment:
I assume this is coming in another diff - but why the change to absolute file paths and outputting as JSON?

Reply:
So:
- outputting as json is more robust
- the get-changed-files action produces files relative to the github base by default; I [added the absolute option](https://github.com/jitterbit/get-changed-files/pull/40) because that's what the rest of our system expects, and it allows us to run commands from various subdirectories, not just the workspace root.